src/app/submissions/submissions.service.ts

```diff
@@ -91,6 +91,9 @@ export class SubmissionsService {
     const oldAnswer = submission.answers[index];
     submission.answers[index] = this.newAnswer(answer, oldAnswer, correct);
     const nextQuestion = this.nextQuestion(submission, index, 'passed');
+    if (submission.answers.findIndex(ans => ans.grade === undefined) === -1) {
```

Comment:
@YuserahN To explain a little more detail: the linting rule that was broken here was for a "shadowed variable".  Specifically, there is a variable `answer` defined as a function parameter on line 89.  In the findIndex if you declare another `answer` variable within that function scope there could be confusion over which `answer` variable is being referred to.

Here's the specific documentation for the rule.  Let us know if you have any more questions!

https://palantir.github.io/tslint/rules/no-shadowed-variable/